pkg/ingestor/collector/collector.go

```diff
@@ -15,4 +15,77 @@
 
 package collector
 
-type Collector interface{}
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/ingestor/processor"
+	"github.com/sirupsen/logrus"
+)
+
+type Collector interface {
+	// Is the collector enabled
+	Enabled(ctx context.Context) bool
+	// Retrieve the artifacts from the collector
+	RetrieveArtifacts(ctx context.Context) ([]*processor.Document, error)
```

Comment:
hmm wondering if this should return a channel to simulate a stream

Reply:
done.